### PR TITLE
[flutter_tools] reland migrate FlutterViews to package:vm_service

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Move FlutterView and related RPCs to the package:vm_service implementation. Update some getIsolate calls with catchError to match previous behavior.

- Updates tests that were previously mocking FlutterViews to use real views
- Moves the FlutterView cache from VM to FlutterDevice
- Catch SentinelException during Isolate.kill
#52179

Maybe #55552

Reland of #55341

### CODE REVIEW

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -233,18 +233,25 @@ class FlutterDevice {
...
       return <FlutterView>[];
     }
-
-
-    return (viewFilter != null
-        ? flutterDeprecatedVmService.vm.allViewsWithName(viewFilter)
-        : flutterDeprecatedVmService.vm.views).toList();
+    if (viewFilter != null) {
+      return <FlutterView>[
+        for (final FlutterView flutterView in _views)
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/resident_runner.dart:249]: This line has the fix, `views` caused a stack overflow.

File: packages/flutter_tools/test/general.shard/resident_runner_test.dart [hunk 2]
```diff
@@ -134,11 +170,35 @@ void main() {
       final Completer<void> result = Completer<void>.sync();
       return result.future;
     });
-    when(mockIsolate.reload()).thenAnswer((Invocation invocation) {
-      return Future<ServiceObject>.value(null);
-    });
   });
 
+  test('FlutterDevice can list views with a filter', () => testbed.run(() async {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/resident_runner_test.dart:175]: I added this test covers to cover the view filter

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
3rd times the charm!

**Pull request merged**